### PR TITLE
fixed /common/example cow_compositions.cpp clone the inner column when there is no need

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed the /common/examples/cow_compositions.cpp, it will not clone the inner column when there is no need

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (alexey-milovidov) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: Wrong modification of the template.

Comment (zombee0) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: > Wrong modification of the template.

so sorry， i am newer for clickhouse，when i use the common/examples/cow_composition.cpp， i found that the inner （wrapped）column would be cloned when there is no need，and fixed it，so what should i do，how can i modify the template

Comment (zombee0) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: > Wrong modification of the template.

the bug is presented in:
https://github.com/ClickHouse/ClickHouse/issues/36449#issue-1209576330

Comment (vdimir) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: @zombee0 just leave https://github.com/ClickHouse/ClickHouse/pull/36453/commits/efae3dce1b6088f79c26f253066c3391f572e83c in this PR, you don't need to edit `PULL_REQUEST_TEMPLATE.md` at all, just set correct description in this PR. Or you can enable edit access for this PR ([see](https://docs.github.com/en/enterprise-server@3.4/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests)) and I'll push directly to your branch.

Comment (zombee0) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: I had "Allow edits by maintainers"

Comment (zombee0) [reply .github/PULL_REQUEST_TEMPLATE.md]: and, I sync the ClickHouse master, and rollback to efae3dc, is this ok?

**Review by vdimir — APPROVED** [review]
Will merge after CI pass (at least build check)

**Comment by zombee0** [comment]
there are 4 PullRequestCI failed, what should i do?

**Comment by vdimir** [comment]
> there are 4 PullRequestCI failed, what should i do?

I've took a look, failures not relate to changes

**Pull request merged**
